build(property-binder): use fluid-tsc and Node16 resolution

DLehenbauer opened the pull request:
Semi-automatic PR to switch to fluid-tsc

Only test CJS as most tests are .js and are using ESM compatible patterns.

DLehenbauer committed:
build(property-binder): use fluid-tsc

DLehenbauer committed:
Merge branch 'main' of https://github.com/microsoft/FluidFramework into property-binder

RishhiB committed:
remove mocha from tsconfig and fix import

RishhiB committed:
add passWithNoTests and undo removing mocha

RishhiB committed:
remove mocha

RishhiB committed:
Merge branch 'main' into property-binder

RishhiB commented on experimental/PropertyDDS/packages/property-binder/package.json:
```diff
@@ -42,8 +56,8 @@
 		"test:coverage": "c8 npm test",
-		"test:jest": "jest",
-		"tsc": "tsc"
+		"test:jest": "jest --passWithNoTests",
```
added --passWithNoTests

jason-ha replied:
There were no tests because the tsconfig.json settings were changed and removed "allowJs". Restored and now there are tests again. :)

RishhiB commented on experimental/PropertyDDS/packages/property-binder/src/test/tsconfig.json:
```diff
@@ -1,12 +1,9 @@
 		"rootDir": "./",
-		"outDir": "../../dist/test",
+		"outDir": "../../lib/test",
+		"types": ["node"],
```
mocha was added to this, removed

RishhiB commented on experimental/PropertyDDS/packages/property-binder/src/internal/activationQueryCacheHelper.ts:
```diff
@@ -6,9 +6,9 @@ import { PropertyFactory } from "@fluid-experimental/property-properties";
 import * as _ from "underscore";
-import { getLocalOrRemoteSchema } from "../data_binder/internalUtils";
-import { DataBinder, DataBinding } from "..";
-import { ExtractedContext } from "@fluid-experimental/property-changeset/dist/helpers/typeidHelper";
```
removed reaching into the package

jason-ha committed:
Merge branch 'main' into property-binder

jason-ha committed:
fix(client-property-binder): formatting

jason-ha committed:
test(client-property-binder): restore testing
-,restore test build settings to allow .js tests to be in place
- remove ESM test build as tests use unsupported patterns

jason-ha committed:
build(client-property-binder): fix build:test
